### DISCUSSION TITLE
Won't recognise item id and deposit into chest

### DISCUSSION THREAD

**Discussion by Wierdwizard2587** [PrismarineJS/mineflayer#2746] [Q&A]
hey all, am making a mining bot and on command the bot will return to a predesignated spot, open the nearest chest and deposit the diamonds it mined into the chest. 

only the Chest opens but it doesn't recognize the diamond id number and won't deposit them.
I've started learning Javascript specifically for minelayer and don't have much experience but can't figure it out. 

been stuck on this one for a long time so any help would be much appreciated

`const mineflayer = require('mineflayer')
const { pathfinder, Movements } = require('mineflayer-pathfinder')
const { GoalGetToBlock } = require('mineflayer-pathfinder').goals
const toolPlugin = require('mineflayer-tool').plugin
var vec3 = require('vec3');
botOptions = {
    host: 'localhost',
    port: 25565,
    username: "Dmd_MinerBot_1"
}

const bot = mineflayer.createBot(botOptions)
bot.loadPlugin(pathfinder)
bot.loadPlugin(toolPlugin)
async function findAndMine(id){
    const block = bot.findBlock({ matching: id, maxDistance: 50000 })
    if (block){
        const promiseHack = new Promise((res,rej)=>{
            bot.once('goal_reached', (goal) => {
                if (!bot.canDigBlock(block)){
                    throw new Error("unable to reack block")
                }
                bot.tool.equipForBlock(block)
                bot.dig(block,false)
                bot.once("diggingCompleted",(block)=>{
                    res()
                })
             })
             bot.pathfinder.setGoal(new GoalGetToBlock(block.position.x, block.position.y, block.position.z))
         })
         await promiseHack
         console.log("promise")
         return
    } else {
        throw new Error("no block found")
    }
}
bot.on("diggingAborted",(block)=>{
    console.log("diggingAborted")
})
bot.on("diggingCompleted",(block)=>{
    console.log("diggingCompleted")
})

function putaway() {

}

bot.on('whisper', (username, message, rawMessage) => {
    console.log("i got the whisper")
    if (message === 'mine d') {
        const mcData = require('minecraft-data')(bot.version)
        const moves = new Movements(bot, mcData)
        bot.pathfinder.setMovements(moves)

        const blockName = "diamond_ore"

        const id = mcData.blocksByName[blockName].id

        findAndMine(id)
        .catch(function(err){
            console.log(err)
        })
        .then(function(){
            console.log("done")
            bot.whisper('Director_Bot', 'mined d')
        });
    }
})

bot.on('whisper', (username, message, rawMessage) => {
    if (message === 'deposit') {
        const mcData = require('minecraft-data')(bot.version);
        const movements = new Movements(bot, mcData);
        movements.scafoldingBlocks = [];
        bot.pathfinder.setMovements(movements);
    
    
        const x = -193;
        const y = 69;
        const z =  -26;
        bot.pathfinder.setGoal(new GoalGetToBlock(x, y, z));

        bot.once('goal_reached', (goal) => {
            bot.openChest(bot.blockAt(vec3(-193,68,-26)));
            depositItem();
         })

    }
})

async function depositItem () {
    let diamond = bot.inventory.findInventoryItem(264);
    let amount = bot.inventory.count('diamond')
    bot.chat(amount)
    if (diamond) {
      try {
        await chest.deposit(264, null, amount)
        bot.chat(`deposited`)
      } catch (err) {
        bot.chat(`unable to deposit`)
      }
    } else {
      bot.chat(`unknown item`)
    }

  }

    
`

Thanks for any Help!

**Comment by amoraschi** [comment]
Use mcData or search by name instead of hardcoding IDs
